src/main/java/org/apache/mesos/scheduler/plan/DefaultBlock.java

```diff
@@ -17,24 +20,23 @@
     private final String name;
     private final Optional<OfferRequirement> offerRequirementOptional;
     private final UUID id = UUID.randomUUID();
+    private final List<String> errors;
+    private final Strategy strategy = new ParallelStrategy();
     private Status status;
     private Map<Protos.TaskID, Status> tasks = new HashMap<>();
 
-    public DefaultBlock(String name, OfferRequirement offerRequirement, Status status) {
-        this(name, Optional.of(offerRequirement), status);
-    }
-
-    private DefaultBlock(String name, Optional<OfferRequirement> offerRequirementOptional, Status status) {
+    public DefaultBlock(
+            String name,
+            Optional<OfferRequirement> offerRequirement,
+            Status status,
+            List<String> errors) {
         this.name = name;
-        this.offerRequirementOptional = offerRequirementOptional;
+        this.offerRequirementOptional = offerRequirement;
         this.status = status;
+        this.errors = errors;
     }
 
-    private boolean isStatus(Status status) {
-        return status.equals(this.status);
-    }
-
-    private void setTaskIds(Collection <Protos.Offer.Operation> operations) {
+    private synchronized void setTaskIds(Collection <Protos.Offer.Operation> operations) {
```

Comment:
Maybe specify in the class' javadoc that it's expected to be called from multiple threads, and specify which threads those would be. I'm assuming it's something like:
- Main execution thread
- Web service/Resource thread
- others?